From 8a24527e6c63914b838698ed78c44cb8a189129a Mon Sep 17 00:00:00 2001
From: Samuel Williams <sam8641@gmail.com>
Date: Sat, 10 May 2025 19:53:09 -0500
Subject: [PATCH] wifi: mt76: mt7921: add 160 MHz AP for mt7922 device

This allows mt7922 in hostapd mode to transmit up to 1.4 Gbps.

Signed-off-by: Samuel Williams <sam8641@gmail.com>
Link: https://patch.msgid.link/20250511005316.1118961-1-sam8641@gmail.com
Signed-off-by: Felix Fietkau <nbd@nbd.name>
---
 mt7921/main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mt7921/main.c b/mt7921/main.c
index 982064088..cbeeda63b 100644
--- a/mt7921/main.c
+++ b/mt7921/main.c
@@ -83,6 +83,11 @@ mt7921_init_he_caps(struct mt792x_phy *phy, enum nl80211_band band,
 			he_cap_elem->phy_cap_info[9] |=
 				IEEE80211_HE_PHY_CAP9_TX_1024_QAM_LESS_THAN_242_TONE_RU |
 				IEEE80211_HE_PHY_CAP9_RX_1024_QAM_LESS_THAN_242_TONE_RU;
+
+			if (is_mt7922(phy->mt76->dev)) {
+				he_cap_elem->phy_cap_info[0] |=
+					IEEE80211_HE_PHY_CAP0_CHANNEL_WIDTH_SET_160MHZ_IN_5G;
+			}
 			break;
 		case NL80211_IFTYPE_STATION:
 			he_cap_elem->mac_cap_info[1] |=
